Expensify/App
# Improve search functions to match multiple names

cc @NicMendonca 

### Details

- Splitted search value into the words. 
- Test regex for each word.
- Removed redundant regex patterns.
- Added `sortByLastMessageTimestamp` to search options.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify.cash/issues/1283

### Tests

- Click seach icon
- Enter "test1 test7" value to search input box
- Make sure the group with test1@example.com and test7@example.com is listed
- Verified that search results are ordered by last message timestamp

### Tested On


### Screenshots

#### Web
<img width="500" alt="Screenshot 2021-02-14 at 17 03 59" src="https://user-images.githubusercontent.com/3853003/107883605-cc5bc880-6ee7-11eb-858a-d797b6c1e5d4.png">

<img width="500" alt="Screenshot 2021-02-14 at 17 04 20" src="https://user-images.githubusercontent.com/3853003/107883611-d251a980-6ee7-11eb-8629-e9f9df015720.png">

#### iOS
<img width="300" alt="Simulator Screen Shot - iPhone X - 2021-02-14 at 17 20 47" src="https://user-images.githubusercontent.com/3853003/107883804-11342f00-6ee9-11eb-9dc6-2d2020571326.png">

<img width="300" alt="Simulator Screen Shot - iPhone X - 2021-02-14 at 17 20 53" src="https://user-images.githubusercontent.com/3853003/107883809-16917980-6ee9-11eb-9f57-c4715b6caa07.png">
